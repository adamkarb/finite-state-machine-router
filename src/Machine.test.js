import React from 'react';
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';
import { Link, Machine, State, Transition } from '.';

describe('<Machine/>', () => {
    const MachineSimple = <Machine id='home'>
        <State id='child'/>
    </Machine>;
    const Child1 = (props) => <div>
        <Link event='link-event'>Link Event</Link>
        <Link href='/home'>Link Event</Link>
    </div>
    const MachineComplex = <Machine id='home'>
        <State id='parent'>
            <State id='child-1' component={Child1}>
                <Transition event='test-event' target='child-2'/>
                <State id='grand-child'/>
            </State>
            <State id='child-2'/>
        </State>
        <State id='parent-2'>
            <State id='child-3'/>
        </State>
    </Machine>;

    const MachineWithPaths = <Machine id='home' path='/home'>
        <State id='parent' path='/:parent'>
            <State id='child-1' path='/child-1' component={Child1}>
                <Transition event='test-event' target='child-2'/>
                <State id='grand-child'/>
            </State>
            <State id='child-2' component={(props) => <div>Child 1</div>}/>
        </State>
        <State id='parent-2'>
            <State id='child-3'/>
        </State>
    </Machine>;

    test('Renders the minimum necessary components for a valid <Machine/>', () => {
        const wrapper = renderer.create(MachineSimple);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('Renders <Machine/> content that does not contain any URLs', () => {
        const wrapper = renderer.create(MachineComplex);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test.skip('Renders <Machine/> content at \'/\'', () => {
        const initialUrl = '/';
        const testHistory = createMemoryHistory({ initialEntries: [ initialUrl ] });
        const wrapper = renderer.create(<Machine history={testHistory} id='home' path={initialUrl}>
            <State id='child-1' path='/child-1' component={(props) => <div>
                <Link event='link-event'>Link Event</Link>
                <Link href='/home'>Link Event</Link>
            </div>}>
                <State id='grand-child'>
                    <State id='great-grand-child'/>
                </State>
            </State>
            <State id='child-2' component={(props) => <h1>Child 2</h1>} path='/child-2'/>
        </Machine>);

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});