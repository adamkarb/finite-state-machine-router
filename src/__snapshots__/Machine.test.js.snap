// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Machine/> Renders <Machine/> content that does not contain any URLs 1`] = `
<div>
  <a
    className={null}
    href="#"
    onClick={[Function]}
  >
    Link Event
  </a>
  <a
    className={null}
    href="/home"
    onClick={[Function]}
  >
    Link Event
  </a>
</div>
`;

exports[`<Machine/> Renders the minimum necessary components for a valid <Machine/> 1`] = `null`;
